StreamChatClient_v3/APIClient/Endpoints/UserEndpoints.swift

```diff
@@ -0,0 +1,164 @@
+        mutedUsers = try container.decodeIfPresent([MutedUser<ExtraData>].self, forKey: .mutedUsers) ?? []
+        teams = try container.decodeIfPresent([String].self, forKey: .teams) ?? []
+        extraData = try? ExtraData(from: decoder)
+        
+        unreadChannelsCount = try container.decodeIfPresent(Int.self, forKey: .unreadChannelsCount)
+        unreadMessagesCount = try container.decodeIfPresent(Int.self, forKey: .unreadMessagesCount)
+    }
+    
+    internal init(
+        id: String, created: Date, updated: Date, lastActiveDate: Date?, isOnline: Bool, isInvisible: Bool, isBanned: Bool,
+        roleRawValue: String, extraData: ExtraData?, devices: [Device], mutedUsers: [MutedUser<ExtraData>],
+        unreadChannelsCount: Int?,
+        unreadMessagesCount: Int?, teams: [String]
+    ) {
+        self.id = id
+        self.created = created
+        self.updated = updated
+        self.lastActiveDate = lastActiveDate
+        self.isOnline = isOnline
+        self.isInvisible = isInvisible
+        self.isBanned = isBanned
+        self.roleRawValue = roleRawValue
+        self.extraData = extraData
+        self.devices = devices
+        self.mutedUsers = mutedUsers
+        self.unreadChannelsCount = unreadChannelsCount
+        self.unreadMessagesCount = unreadMessagesCount
+        self.teams = teams
+    }
+    
+    func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(id, forKey: .id)
+//
+//    id = try container.decode(String.self, forKey: .id)
+//    roleRawValue = try container.decode(String.self, forKey: .role)
+//    created = try container.decode(Date.self, forKey: .created)
+//    updated = try container.decode(Date.self, forKey: .updated)
+//    lastActiveDate = try container.decodeIfPresent(Date.self, forKey: .lastActiveDate)
+//    isOnline = try container.decode(Bool.self, forKey: .isOnline)
+//    isInvisible = try container.decodeIfPresent(Bool.self, forKey: .isInvisible) ?? false
+//    self.isBanned = try container.decodeIfPresent(Bool.self, forKey: .isBanned) ?? false
+//    self.devices = try container.decodeIfPresent([Device].self, forKey: .devices) ?? []
+//    self.mutedUsers = try container.decodeIfPresent([MutedUser<ExtraData>].self, forKey: .mutedUsers) ?? []
+//    self.teams = try container.decodeIfPresent([String].self, forKey: .teams) ?? []
+//    self.extraData = try? ExtraData(from: decoder)
+//
+//    self.unreadChannelsCount = try container.decodeIfPresent(Int.self, forKey: .unreadChannelsCount)
+//    self.unreadMessagesCount = try container.decodeIfPresent(Int.self, forKey: .unreadMessagesCount)
+//
+    }
+}
+
+extension UserModel {
+    convenience init(endpointResponse: UserEndpointPayload<ExtraData>) {
+        self.init(id: endpointResponse.id, extraData: endpointResponse.extraData)
+    }
+}
+
+/// A muted user.
+struct MutedUser<ExtraData: UserExtraData>: Decodable {
```

Comment:
I thought we were going to do `MutedUser: User`?

Reply:
This whole `MutedUser` part is `TODO`, I forgot to mark it.

Reply:
fixed in 94cf747